### PR TITLE
Password sign in via javascript

### PR DESCRIPTION
**Pull request by walaura**
Sign in (2 step) with a password via javascript too, making the flow neater and more lively.

screen shot 2018-03-29 at 4 04 29 pm

### CODE REVIEW

**Comment by Calum-Campbell** [comment]
I think this needs to change for the new post endpoint on this form

File: public/components/two-step-signin/two-step-signin__slide.js [hunk 1]
```diff
@@ -10,10 +11,14 @@ const SLIDE_STATE_DEFAULT: string = 'SLIDE_STATE_DEFAULT';
 
 const EV_DONE: string = 'form-done';
 
-const ERR_MALFORMED_HTML: string = 'Something went wrong';
-const ERR_MALFORMED_RESPONSE: string = 'Something went wrong';
+const ERR_MALFORMED_HTML: string = 'Something went wrong (html)';
```
Comment (Calum-Campbell) [inline public/components/two-step-signin/two-step-signin__slide.js]: I would prefer this to say "malformed html", but it doens't seem like it's used anyway so maybe can get rid?

Comment (walaura) [reply public/components/two-step-signin/two-step-signin__slide.js]: yeah, the error reporting is a bit of a mess but ideally this _may_ end up being user facing, for now this is mostly so we ca pick them up on sentry

File: app/com/gu/identity/frontend/controllers/SigninAction.scala [hunk 2]
```diff
@@ -134,7 +134,12 @@ class SigninAction(
       case Right(cookies) => Right {
         if (stage == "PROD") Tip.verify("Account Signin")
         metricsLogger(request)
-        successResponse(successfulReturnUrl, cookies)
+        if(request.headers.toSimpleMap.contains("x-gu-browser-rq")){
+          successAjaxResponse(successfulReturnUrl, cookies)
+        }
```
Comment (rahilb) [inline app/com/gu/identity/frontend/controllers/SigninAction.scala]: the formatting is wrong, it should be 

`} else {`

File: app/com/gu/identity/frontend/controllers/SigninAction.scala [hunk 3]
```diff
@@ -168,6 +173,12 @@ class SigninAction(
     SeeOther(successfulReturnUrl.url)
       .withCookies(cookies: _*)
 
+
+  def successfulAjaxSignInResponse(successfulReturnUrl: ReturnUrl, cookies: Seq[Cookie]): Result =
+    Ok("{\"status\": true, \"returnUrl\": \"" + successfulReturnUrl.url + "\"}")
```
Comment (rahilb) [inline app/com/gu/identity/frontend/controllers/SigninAction.scala]: you can use triple quotes `"""{"foo":"bar"}"""` and then you don't need to do this escaping.

Comment (rahilb) [reply app/com/gu/identity/frontend/controllers/SigninAction.scala]: Aditionally, this is prone to error if we have bad values (won't be escaped)... we should always construct json in a typesafe way using the AST.

**Pull request merged**